matron/src/hardware/screen.c

```diff
@@ -449,5 +449,36 @@ extern void screen_export_png(const char *s) {
     cairo_surface_write_to_png(surface, s);
 }
 
+void screen_set_operator(int i) {
+    CHECK_CR
+    cairo_operator_t a[24];
```

Comment:
the base type of the `cairo_operator_t` enum is an int, it may be more straightforward to cast `i` to the appropriate
type (after ensuring i contains a valid value)

Reply:
> it may be more straightforward to cast i to the appropriate
type (after ensuring i contains a valid value)

Wouldn't this mean that all operators would have to be included then? There are some not in the array, such as `CAIRO_OPERATOR_HSL_SATURATION` which have no effect on grayscale values.

Reply:
Perhaps one option is to define a lookup table for the blend modes so that the "public" API lets you call them by name:

```lua
Screen.BLEND_MODES = {
  ['CLEAR'] = 0,
  ['SOURCE'] = 1,
  ...
}

Screen.blend_mode = function(mode)
  _norns.screen_blend_mode(Screen.BLEND_MODES[mode])
end
```

Then the int passed to the `_screen_blend_mode` weaver function could be cast to a `cairo_operator_t` and as a script author I would call `screen.blend_mode('SOURCE')` instead of remembering the numbering or defining my own lookup table.

Reply:
we have a similar issue regarding fonts, which is just a blind index (and the font names are (of minor assistance) reported at matron startup.

just to say, it'd be nice to set up a pattern to follow for similar cases